vault/identity_store_structs.go

```diff
@@ -0,0 +1,184 @@
+package vault
+
+import (
+	"regexp"
+	"time"
+
+	memdb "github.com/hashicorp/go-memdb"
+	"github.com/hashicorp/vault/helper/locksutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	log "github.com/mgutz/logxi/v1"
+)
+
+const (
+	// Storage prefixes
+	entityPrefix = "entity/"
+)
+
+var (
+	// metaKeyFormatRegEx checks if a metadata key string is valid
+	metaKeyFormatRegEx = regexp.MustCompile(`^[a-zA-Z0-9_-]+$`).MatchString
```

Comment:
I think it'd be really neat if we also allowed `=`, `+`, and `/`. Then you can actually use any key you want by simply base64-encoding it!